src/pages/task/ActionButtons.tsx

```diff
@@ -0,0 +1,248 @@
+    ScheduleTaskMutation,
+    ScheduleTaskMutationVariables
+  >(SCHEDULE_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task marked as scheduled");
+    },
+    onError: (err) => {
+      error(`Error scheduling task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [unscheduleTask, { loading: loadingUnscheduleTask }] = useMutation<
+    UnscheduleTaskMutation,
+    UnscheduleTaskMutationVariables
+  >(UNSCHEDULE_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task marked as unscheduled");
+    },
+    onError: (err) => {
+      error(`Error unscheduling task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [abortTask, { loading: loadingAbortTask }] = useMutation<
+    AbortTaskMutation,
+    AbortTaskMutationVariables
+  >(ABORT_TASK, {
+    variables: {
+      taskId,
+    },
+    onCompleted: () => {
+      success("Task aborted");
+    },
+    onError: (err) => {
+      error(`Error aborting task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [restartTask, { loading: loadingRestartTask }] = useMutation<
+    RestartTaskMutation,
+    RestartTaskMutationVariables
+  >(RESTART_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task scheduled to restart");
+    },
+    onError: (err) => {
+      error(`Error restarting task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [setTaskPriority, { loading: loadingSetPriority }] = useMutation<
+    SetTaskPriorityMutation,
+    SetTaskPriorityMutationVariables
+  >(SET_TASK_PRIORTY, {
+    onCompleted: (data) => {
+      success(`Priority for task updated to ${data.setTaskPriority.priority}`);
+    },
+    onError: (err) => {
+      error(`Error updating priority for task: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const disabled =
+    loadingAbortTask ||
+    loadingRestartTask ||
+    loadingSetPriority ||
+    loadingUnscheduleTask ||
+    loadingScheduleTask;
+
+  useEffect(() => {
+    if (disabled) {
+      setIsVisible(false);
+    }
+  }, [disabled]);
+
+  useOnClickOutside(wrapperRef, () => {
+    if (
+      !get(priorityRef, "current.className", "").includes("ant-popover-open")
+    ) {
+      setIsVisible(false);
+    }
+  });
+
+  const rowButtons = [
+    <Button
+      size="small"
+      dataCy="schedule-task"
+      key="schedule"
+      disabled={disabled || !canSchedule}
+      loading={loadingScheduleTask}
+      onClick={scheduleTask}
+    >
+      Schedule
+    </Button>,
+    <Button
+      size="small"
+      dataCy="restart-task"
+      key="restart"
+      disabled={disabled || !canRestart}
+      loading={loadingRestartTask}
+      onClick={restartTask}
+    >
+      Restart
+    </Button>,
+    <Button
+      size="small"
+      dataCy="notify-task"
+      key="notifications"
+      disabled={disabled}
+    >
+      Add Notification
+    </Button>,
+  ];
+
+  const cardItems = [
```

Comment:
i'm so confused about the role `card` plays in all of this

Reply:
Card items are the dropdown items we see in the CardDropdownButton. I should rename CardDropdownButton to DropdownButton. I see why this prop is confusing because it's being passed down two layers. To make this simpler, I can remove ButtonRow (which imports CardDropdownButton) and save the styles it uses to our shared styles.